### PR TITLE
Implement the markAsDirty form controls for AddEditComponent(#761).

### PR DESCRIPTION
**Pull request by gryffs**
This PR should correct the issue reported in #761 

The issue was due to the fact the the password generator filled the new password but did not mark the input as dirty. So, if the input was still marked as pristine the prompt would not trigger. I added a custom method for marking the password input as dirty and called said method when the password is changed.

This issue also existed if the user attempted to add a new login and had not touched any field other than the password generator and this also corrects the functionality there. Please let me know what you think.

### CODE REVIEW

**Comment by gryffs** [comment]
This issue is also reported here: #563

File: src/app/vault/add-edit.component.ts [hunk 1]
```diff
@@ -84,4 +89,8 @@ export class AddEditComponent extends BaseAddEditComponent implements OnChanges,
         return (!this.editMode || this.cloneMode) && this.ownershipOptions
             && (this.ownershipOptions.length > 1 || !this.allowPersonal);
     }
+
+    markPasswordAsDirty() {
+        this.form.controls['Login.Password'].markAsDirty();
+      }
```
Comment (cscharf) [inline src/app/vault/add-edit.component.ts:95]: Just a small formatting nit (remove extra indentation, aligning under `markPasswordAsDirty`.
```suggestion
    }
```

**Review by cscharf — COMMENTED** [review]
Thanks @gryffs ! I _believe_ this looks good aside from the 1 formatting nit, although I would like @eliykat to take a look and just validate everything looks good for final approval.

**Review by eliykat — COMMENTED** [review]
Looks good to me, I think this can be approved & merged once @cscharf's suggestion is committed. Thanks @gryffs !

**Comment by gryffs** [comment]
Thanks @cscharf & @eliykat ! I'll make that change in the next day or so and push it up.

File: src/app/vault/add-edit.component.ts (hunk 1 above)
Comment (gryffs) [reply src/app/vault/add-edit.component.ts]: I've corrected the above issue.

**Comment by gryffs** [comment]
@cscharf & @eliykat The formatting issue has been resolved. Let me know if there is anything else. Happy to help!

Here it is in action:

https://user-images.githubusercontent.com/16548994/116777157-81416280-aa9f-11eb-8ca0-c82f33d16188.mov

**Review by eliykat — APPROVED** [review]

**Pull request merged**